eland/field_mappings.py

```diff
@@ -162,8 +183,16 @@ def flatten(x, name=''):
         for index in mappings:
             if 'properties' in mappings[index]['mappings']:
                 properties = mappings[index]['mappings']['properties']
-
-                flatten(properties)
+            else:
+                # Pre Elasticsearch 7.0 mappings had types. Support these
+                # in case eland is connected to 6.x index - this is not
+                # officially supported, but does help usability
+                es_types = list(mappings[index]['mappings'].keys())
+                if len(es_types) != 1:
+                    raise NotImplementedError("eland only supports 0 or 1 Elasticsearch types")
```

Comment:
Maybe this exception can be augmented to add the `es_types` field to give the user an indication what we're seeing on our end?